### PR TITLE
Add Load Balance waiters and examples

### PR DESCRIPTION
**Pull request by joaopalet**

### CODE REVIEW

File: examples/loadbalancer/loadbalancer.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	loadbalancerClient, err := loadbalancer.NewAPIClient(
+		config.WithRegion("eu01"),
+	)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Creating API client: %v\n", err)
+		os.Exit(1)
+	}
+
+	// List the load balancer instances for your project
+	listInstancesResp, err := loadbalancerClient.ListLoadBalancers(context.Background(), projectId).Execute()
```
Comment (vicentepinto98) [inline examples/loadbalancer/loadbalancer.go:27]: Dont you need to enable the LoadBalancer first?

File: examples/loadbalancer/loadbalancer.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+				Targets: &[]loadbalancer.Target{
+					{
+						DisplayName: utils.Ptr("example-target"),
+						Ip:          utils.Ptr("192.0.2.5"),
+					},
+				},
+			},
+		},
+	}
+	createLoadBalancerRes, err := loadbalancerClient.CreateLoadBalancer(context.Background(), projectId).CreateLoadBalancerPayload(createLoadBalancerPayload).XRequestID("X-Request-ID").Execute()
```
Comment (vicentepinto98) [inline examples/loadbalancer/loadbalancer.go:72]: does it work with this `X-Request-ID`? I thought maybe the API checks if it is an UUID

File: services/loadbalancer/wait.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	}
+	// Some APIs may return temporary errors and the request should be retried
+	if utils.Contains(wait.RetryHttpErrorStatusCodes, oapiErr.statusCode) {
+		return nil, false, nil
+	}
+	return nil, false, reqErr
+}
+
+// CreateInstanceWaitHandler will wait for creation
+func CreateInstanceWaitHandler(ctx context.Context, a APIClientInterface, projectId, instanceName string) *wait.Handler {
```
Comment (vicentepinto98) [inline services/loadbalancer/wait.go:50]: No waiter needed for the UpdateTargetPools?

**Review by vicentepinto98 — CHANGES_REQUESTED** [review]

File: examples/loadbalancer/loadbalancer.go (hunk 1 above)
Comment (joaopalet) [reply examples/loadbalancer/loadbalancer.go:35]: Not sure, but moved to start

File: services/loadbalancer/wait.go (hunk 3 above)
Comment (joaopalet) [reply services/loadbalancer/wait.go:50]: Discussed offline, will confirm with team and add later if needed

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**